### PR TITLE
Wrong redirect route?

### PR DESCRIPTION
**Pull request by auipga**
I am targetting this branch, because…

Closes #
Fixes #
## Changelog

``` markdown
### Added
- Added some `Class::newMethod` to do great stuff

### Changed

### Deprecated

### Removed

### Fixed

### Security
```
## To do

- [ ] Update the tests
- [ ] Update the documentation
- [ ] Add an upgrade note
## Subject

Seems more valid to me.
If I'm wrong, can somebody explain this?

### CODE REVIEW

**Comment by greg0ire** [comment]
Why didn't you fill in the PR template?

**Comment by core23** [comment]
Please update the PR template, so we can merge this @auipga

File: Controller/RegistrationFOSUser1Controller.php [hunk 1]
```diff
@@ -56,10 +56,10 @@ public function registerAction()
                 $url = $this->generateUrl('fos_user_registration_check_email');
             } else {
                 $authUser = true;
-                $route = $this->get('session')->get('sonata_basket_delivery_redirect');
+                $route = $this->get('session')->get('sonata_user_redirect_url');
 
                 if (null !== $route) {
-                    $this->get('session')->remove('sonata_basket_delivery_redirect');
+                    $this->get('session')->remove('sonata_user_redirect_url');
```
Comment (greg0ire) [inline Controller/RegistrationFOSUser1Controller.php]: It looks indeed weird… It was introduced in 4dbf95fc5a103eaa364abfd3ce3d805e5f7dcda8

Comment (greg0ire) [inline Controller/RegistrationFOSUser1Controller.php]: @eko , can you please comment on this?

File: Controller/RegistrationFOSUser1Controller.php [hunk 2]
```diff
@@ -56,10 +56,10 @@ public function registerAction()
...
                     $url = $this->generateUrl($route);
                 } else {
                     $url = $this->get('session')->get('sonata_user_redirect_url', $this->generateUrl('sonata_user_profile_show'));
```
Comment (greg0ire) [inline Controller/RegistrationFOSUser1Controller.php]: This is very very similar. @auipga , maybe we should only keep the else part, right?

Comment (greg0ire) [inline Controller/RegistrationFOSUser1Controller.php]: Let's translate in pseudo-code : 

```
get the user redirect url in session

if it is not null
	remove it from the session
	generate an url from it (so it is in fact a route, not an url)
else
	try to get it again anyway, and default to the profile show route
```

Sound pretty stupid, right?  :stuck_out_tongue: 

Let 's just do 
```php
<?php
$url = $this->get('session')->get(
    'sonata_user_redirect_url',
     $this->generateUrl('sonata_user_profile_show')
);
// and maybe this too
$this->get('session')->remove('sonata_user_redirect_url');
```

File: Controller/RegistrationFOSUser1Controller.php (hunk 1 above)
Comment (eko) [inline Controller/RegistrationFOSUser1Controller.php]: Hi @greg0ire,

Indeed, this was for the `ecommerce` bundle suite in order to let user come back on its basket when he choose to register from the basket page (https://github.com/sonata-project/ecommerce/blob/7e778d57d1229df3f73b453404f9e7360b0ca923/src/BasketBundle/Controller/BasketController.php#L53).

Comment (greg0ire) [inline Controller/RegistrationFOSUser1Controller.php]: Thanks for the feedback, do you agree with my proposal below?

File: Controller/RegistrationFOSUser1Controller.php [hunk 3]
```diff
@@ -56,10 +56,10 @@ public function registerAction()
                 $url = $this->generateUrl('fos_user_registration_check_email');
             } else {
                 $authUser = true;
-                $route = $this->get('session')->get('sonata_basket_delivery_redirect');
+                $route = $this->get('session')->get('sonata_user_redirect_url');
```
Comment (greg0ire) [inline Controller/RegistrationFOSUser1Controller.php]: This is bad, you're storing a url in a route.

**Comment by greg0ire** [comment]
I made [an alternative PR](https://github.com/sonata-project/SonataUserBundle/pull/806). What do you think?

**Pull request closed without merging**